lib/models/package-info-cache/package-info.js

```diff
@@ -5,6 +5,24 @@ const ErrorList = require('./error-list');
 const Errors = require('./errors');
 const AddonInfo = require('../addon-info');
 
+function lexicographically(a, b) {
+  if (typeof a.name === 'string' && typeof b.name === 'string') {
+    return a.name.localeCompare(b.name);
+  }
+  return 0;
+}
+
+// lastWriteWins
```

Comment:
I would consider using a stable sort as well (map items to an array with the index value so you can compare on it) if unnamed addons are a common.
something like
`addons = addons.map((addon, i) => { name: addon.name, i, addon }).sort(compareNameIndex).map(({addon}) => addon);`

Reply:
Un-named addon's ultimately get dropped later, I'll do work to drop them sooner. Thereby removing that from leaking into this code at all.